app/controllers/posts_controller.rb

```diff
@@ -94,6 +98,10 @@ def set_post
   end
   # Never trust parameters from the scary internet, only allow the white list through.
   def post_params
-    params.require(:post).permit(:title, :contents, :author_id)
+    @post_params ||= begin
+     p = params.require(:post).permit(:title, :contents, :author_id, :subjects)
+     p[:subjects] = Subjects.find(p[:subjects]) if p.include? :subjects
```

Comment:
You can merge with this and I'll fix it later.

Basically you are passing a collection of subject ids and this line converts them into an array of Subject objects. But that isn't necessary because you just need the id. So you are wasting time fetching all the objects from the db.